addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,816 @@
+				SetEntPropFloat(client, Prop_Send, "m_flLaggedMovementValue", gf_TimeScale[client]);
+				float fTimescale = GetEntPropFloat(client, Prop_Data, "m_flLaggedMovementValue");
+				gf_TimescaleTicksPassed[client] += fTimescale;
+				truevel[client] = GetClientVelo(client);
+				/*
+							AUTO STRAFER START
+														*/
+				if(buttons & IN_FORWARD && vel[0] <= 50.0)
+					vel[0] = 450.0;
+
+				float yaw_change = 0.0;
+				if(vel[0] > 50.0)
+					yaw_change = 30.0 * FloatAbs(30.0 / vel[0]);
+
+				if (AutoStrafeEnabled[client] == true && Shavit_GetTimerStatus(client) == Timer_Running && g_bTAS[client] && !(GetEntityFlags(client) & FL_ONGROUND) && (GetEntityMoveType(client) != MOVETYPE_NOCLIP) && !(buttons & IN_FORWARD) && !(buttons & IN_BACK) && !(buttons & IN_MOVELEFT) && !(buttons & IN_MOVERIGHT))
+				{
+					if(mouse[0] > 0)
+					{
+						angles[1] += yaw_change;
+						//buttons |= IN_MOVERIGHT;
+						if(g_Engine == Engine_CSS) {
+							vel[1] = 400.0;
+						} else if(g_Engine == Engine_CSGO){
+							vel[1] = 450.0;
+						}
+					}
+					else if(mouse[0] < 0)
+					{
+						angles[1] -= yaw_change;
+						//buttons |= IN_MOVELEFT;
+						if(g_Engine == Engine_CSS) {
+							vel[1] = -400.0;
+						} else if(g_Engine == Engine_CSGO){
+							vel[1] = -450.0;
+						}
+					}
+				}
+				/*
+							AUTO STRAFER END
+														*/
+
+				/*
+							WIGGLEHACK START
+														*/
+				if (g_Strafing[client] == true && Shavit_GetTimerStatus(client) == Timer_Running && g_bTAS[client] && !(GetEntityFlags(client) & FL_ONGROUND) && (GetEntityMoveType(client) != MOVETYPE_NOCLIP) && !(buttons & IN_FORWARD) && !(buttons & IN_BACK) && !(buttons & IN_MOVELEFT) && !(buttons & IN_MOVERIGHT))
+				{
+					if(gf_TimescaleTicksPassed[client] >= 1.0) {
+						//Don't subtract 1 from gf_TimescaleTicksPassed[client] because it happens later and this code won't always run depending on if wiggle hack is on.
+
+						if(AngDiff[client] < AutoStrafeTrigger * -1) {
+							if(g_Engine == Engine_CSS) {
+								vel[StrafeAxis[client]] = -400.0;
+							} else if(g_Engine == Engine_CSGO){
+								vel[StrafeAxis[client]] = -450.0;
+							}
+						}
+						else if(AngDiff[client] > AutoStrafeTrigger) {
+							if(g_Engine == Engine_CSS) {
+								vel[StrafeAxis[client]] = 400.0;
+							} else if(g_Engine == Engine_CSGO){
+								vel[StrafeAxis[client]] = 450.0;
+							}
+						} else if (!(GetEntityFlags(client) & FL_ONGROUND) && (GetEntityMoveType(client) != MOVETYPE_NOCLIP)) {
+							if (!(truevel[client] == 0.0)) {
+								flYawBhop[client] = 0.0;
+								float x = 30.0;
+								float y = truevel[client];
+								float z = x/y;
+								z = FloatAbs(z);
+								flYawBhop[client] = x * z;
+							}
+						
+							if (DirIsRight[client] == true) {
+								angles[1] += flYawBhop[client];
+								//buttons |= ~IN_MOVERIGHT;
+								DirIsRight[client] = false;
+								if(g_Engine == Engine_CSS) {
```

Comment:
Would be better to do this check once when the plugin starts and have a variable that holds the 400/450 in it. no point in checking the engine type 20 times a tick for every player.